apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -37,25 +28,22 @@ import { ListItemComponent } from '../list-item/list-item.component';
     </div>
   `,
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [ListItemComponent],
 })
 export class CardComponent {
   @Input() list: any[] | null = null;
   @Input() type!: CardType;
+  @Input() showingProp!: string;
   @Input() customClass = '';
 
-  CardType = CardType;
-
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore,
-  ) {}
+  constructor(@SkipSelf() private storeService: Store<CardItem>) {}
```

Comment:
you should have no logic inside this component. This component is a presentational component and its only purpose is to display the data. 

Use @Ouput to dispatch event to the parent. 
`ng-content` to project content
and `ngTemplateOulet` to project more complex content.